opwen_email_server/utils/email_parser.py

```diff
@@ -75,6 +82,44 @@ def parse_mime_email(mime_email: str) -> dict:
     }
 
 
+def format_attachments(email: dict) -> dict:
+    attachments = email.get('attachments', [])
+
+    if not attachments:
+        return email
+
+    formatted_attachments = deepcopy(attachments)
+    is_any_attachment_changed = False
+
+    for i, attachment in enumerate(attachments):
+        filename = attachment.get('filename', '')
+        content = attachment.get('content', '')
+        formatted_content = _format_attachment(filename, content)
+
+        if content != formatted_content:
+            formatted_attachments[i]['content'] = formatted_content
+            is_any_attachment_changed = True
+
+    if not is_any_attachment_changed:
+        return email
+
+    new_email = dict(email)
+    new_email['attachments'] = formatted_attachments
+    return new_email
+
+
+def _format_attachment(filename: str, content: str) -> str:
+    guessed_type = guess_type(filename)[0]
+
+    if not guessed_type:
+        return content
+
+    if 'image' in guessed_type.lower():
+        image = _change_image_size(content)
```

Comment:
Nit: it would be nicer to do a "mutation chain" style here instead of an early return since potentially we can have many mutations on the attachments. So like this:

```py
if 'image' in guess_type.lower():
    content = _change_image_size(content)

# ...
# potentially more optimizations here, e.g. reencode jpg images for more size savings
# ...

return content
```